### PR TITLE
RAL modifications against upstream

### PR DESCRIPTION
**Pull request by jrha**
For reference only, DO NOT MERGE.

### CODE REVIEW

**Comment by jrha** [comment]
The only changes here we are particularly attached to are:
- 694fec0 Relax hardware labelling to include dash and underscore
- 3f54716 Allow an additional decimal at the end of an interface name
- 4f32488 Add support for aq search_host --feature
- 32d24c4 Allow group members to access sockdir
- 36adab7 Add support for an archetype preface template

While we need a method of reporting the status of sandboxes but the script in this commit is probably not a great way of achieving that.
- de62961 Add script to produce json report of sandboxes statuses

Everything else is RAL specific.

**Comment by stdweird** [comment]
@jrha oopsie, i clicked on the 'merge conflicts" editor, and it generated a new commit straight in your PR. i expected it to open a PR to your branch...

File: bin/aq [hunk 1]
```diff
@@ -0,0 +1 @@
+/opt/aquilon/bin/aq.py
```
Comment (stdweird) [inline bin/aq:1]: this is an absolute path and cannot be packaged by bdist_rpm.
i propose to replace it with a relative symlink to aq.py (ie in the same directory).

**Comment by jrha** [comment]
:fearful:

**Comment by urbonegi** [comment]
@jrha  - for our internal review process we need the commit log to have a Change-Id: line.

Please can you install this hook in .git/hooks/commit-msg before submitting future requests?

For this commit, please can you then amend the commit log with a Change-Id line? (You can just pick the original commit id) We can't do it because the commit-id will change and GH won't see it as the same commit.

**Comment by jrha** [comment]
Sure, although these probably need splitting up and rebasing seperately on the latest version of the code base, so I'll probably open a new PR.

**Comment by jrha** [comment]
"36adab7 Add support for an archetype preface template" cherry-picked into the new PR #64.

**Comment by jrha** [comment]
Will probably just close this once we've updated our broker - fairly sure the upstream code now contains everything we needed.

**Comment by urbonegi** [comment]
@jrha  can we close this one? The config problem is now solved. As for other code parts that uses MS reference we will need to submit separate change (to move 'ms.com' to a config option when it is hardcoded. ). Basically it is easier to have separate PRs for separate fixes/improvements - we can then review that and merge quicker - not causing long running discussion :)

**Comment by jrha** [comment]
Yes, let's close it.

**Pull request closed without merging**